conwayste/src/ui/pane.rs

```diff
@@ -77,24 +78,37 @@ impl Pane {
 
         // forward_mouse_events
         // TODO: all mouse event types, not just Click
-        let handler = Box::new(|_obj: &mut dyn EmitEvent, uictx: &mut context::UIContext, evt: &context::Event| -> Result<Handled, Box<dyn Error>> {
-            // let pane = obj.downcast_mut::<Pane>()?; // uncomment and rename _obj to obj above if we need a Pane
-
-            for child_id in uictx.widget_view.children_ids() {
-                let (widget_ref, mut subuictx) = uictx.derive(&child_id).unwrap(); // unwrap OK because 1) valid ID, 2) in view
-
-                let point = &evt.point.unwrap(); // unwrap OK because a Click event always has a point
-                if within_widget(&point, &widget_ref.rect()) {
-                    if let Some(emittable_ref) = widget_ref.as_emit_event() {
-                        emittable_ref.emit(evt, &mut subuictx)?;
-                    } else {
-                        warn!("Widget at point of click ({:?}) does not implement EmitEvent", evt.point);
-                    }
-                }
+        for event_type in EventType::into_enum_iter() {
+            if event_type.is_mouse_event() {
+                let handler = Box::new(
+                    |_obj: &mut dyn EmitEvent,
+                     uictx: &mut context::UIContext,
+                     evt: &context::Event|
+                     -> Result<Handled, Box<dyn Error>> {
+                        // let pane = obj.downcast_mut::<Pane>()?; // uncomment and rename _obj to obj above if we need a Pane
+
+                        for child_id in uictx.widget_view.children_ids() {
+                            let (widget_ref, mut subuictx) = uictx.derive(&child_id).unwrap(); // unwrap OK because 1) valid ID, 2) in view
+
+                            let point = &evt.point.unwrap(); // unwrap OK because a Click event always has a point
+                            if within_widget(&point, &widget_ref.rect()) {
+                                if let Some(emittable_ref) = widget_ref.as_emit_event() {
+                                    emittable_ref.emit(evt, &mut subuictx)?;
+                                } else {
+                                    warn!("Widget at point of click ({:?}) does not implement EmitEvent", evt.point);
+                                }
+                            }
+                        }
+                        Ok(Handled::Handled)
+                    },
+                );
```

Comment:
Nitpick, but you don't need the comma on line 102.

If `within_widget()` on line 94 never evaluated to true, why do we  still return `Handled` here on line 102? I'm wondering if we should only return `Handled` if we emitted on something (line 97).

Reply:
That comma was added by rustfmt. I could take it out but if we ever rustfmt our whole project it'll come back. :disappointed: 

And I like your idea. I'll fix this.

Reply:
Ah it's easier to read too. :clap: